google/cloud/ndb/_gql.py

```diff
@@ -715,16 +715,6 @@ def get_query(self):
         ancestor = None
         model_filters = list(model_class._default_filters())
         filters = self.query_filters(model_class, model_filters)
-        offset = self.offset()
-        limit = self.limit()
-        if limit < 0:
-            limit = None
-        keys_only = self.is_keys_only()
-        if not keys_only:
-            keys_only = None
-        default_options = query_module.QueryOptions(
-            offset=offset, limit=limit, keys_only=keys_only
-        )
         projection = self.projection()
```

Comment:
If `offset`, `limit`, and `keys_only` are specified in the GQL, will they get lost?

Reply:
I will have to rethink this part. I missed how the options decorator uses these.

Reply:
Fixed it by sending None when all the options are None.

Reply:
Hmm. That gets the warning out of the tests, but still generates a warning for something a user might do innocently. I'm looking at it now and I'm wondering if I shouldn't have had `offset`, `limit`, and `keys_only` options present on the constructor to begin with? Either that, or maybe deprecating `default_options` was my mistake? Because it does make sense you'd potentially want to set those on the constructor. I, admittedly, wasn't thinking about GQL when I did any of that.

You can see in `QueryOptions`, where I tried to organize somewhat where options would potentially be coming from, and maybe they all three belong in the "both (!?!)" category with `projection`. And with the benefit of time, we can probably drop the astonished "(!?!)" now.

What do you think?

Reply:
Well, I wouldn't mind having those options in the regular ones. I don't see a problem with the deprecation. It's just that we should avoid the accidental deprecation warnings coming from our own code if possible.

Reply:
@chrisrossi so if you are OK with this, I'll add those options.

Reply:
Do you mean adding those arguments to the constructor? That works for me if it works for you!

Reply:
@chrisrossi I added the arguments and updated the tests.